src/Common/Extensions/UrlExtensions.cs

```diff
@@ -0,0 +1,65 @@
+namespace Passwordless.Common.Extensions;
+
+public static class UrlExtensions
+{
+    public static bool IsLocalUrl(this string? url)
```

Comment:
Why not use the `Uri` class?

Reply:
This is literally the code we were using in Razor Pages.

```https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.Core/src/Routing/UrlHelperBase.cs```